File: qt/gen_qrunnable.go
package qt

import (
	"unsafe"
)

type QRunnable struct {
	h          uintptr
	isSubclass bool
}

// NewQRunnable constructs a new QRunnable object.
func NewQRunnable() *QRunnable {
	ret := newQRunnable(QRunnable_new())
	ret.isSubclass = true
	return ret
}

func (this *QRunnable) Run() {
	QRunnable_Run(this.h)
}

func (this *QRunnable) AutoDelete() bool {
	return (bool)(QRunnable_AutoDelete(this.h))
}

func (this *QRunnable) SetAutoDelete(autoDelete bool) {
	QRunnable_SetAutoDelete(this.h, (bool)(autoDelete))
}

type QGenericRunnable struct {
	h          uintptr
	isSubclass bool
}

func (this *QGenericRunnable) Run() {
	QGenericRunnable_Run(this.h)
}

File: qt/gen_qrunnable.cpp
// +build ignore

#include <QGenericRunnable>
#include <QRunnable>
#include <qrunnable.h>
#include "gen_qrunnable.h"

QRunnable* QRunnable_new() {
	return new QRunnable();
}

void QRunnable_Run(QRunnable* self) {
	self->run();
}

bool QRunnable_AutoDelete(const QRunnable* self) {
	return self->autoDelete();
}

void QRunnable_SetAutoDelete(QRunnable* self, bool autoDelete) {
	self->setAutoDelete(autoDelete);
}

void QRunnable_Delete(QRunnable* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QRunnable*>( self );
	} else {
		delete self;
	}
}

void QGenericRunnable_virtbase(QGenericRunnable* src, QRunnable** outptr_QRunnable) {
	*outptr_QRunnable = static_cast<QRunnable*>(src);
}

void QGenericRunnable_Run(QGenericRunnable* self) {
	self->run();
}

void QGenericRunnable_Delete(QGenericRunnable* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QGenericRunnable*>( self );
	} else {
		delete self;
	}
}

